tests/Unit/PointwiseFunctions/GeneralRelativity/Test_ProjectionOperators.cpp

```diff
@@ -0,0 +1,241 @@
+
+#include <cstddef>
+#include <random>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "Domain/CoordinateMaps/Affine.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/ProductMaps.hpp"
+#include "Domain/CoordinateMaps/ProductMaps.tpp"
+#include "Domain/LogicalCoordinates.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Domain/Structure/Side.hpp"
+#include "Domain/Tags.hpp"
+#include "Framework/CheckWithRandomValues.hpp"
+#include "Framework/SetupLocalPythonEnvironment.hpp"
+#include "Helpers/DataStructures/DataBox/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "PointwiseFunctions/GeneralRelativity/ProjectionOperators.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/TagsDeclarations.hpp"
+
+namespace {
+template <size_t SpatialDim, typename DataType>
+void test_projection_operator(const DataType& used_for_size) {
+  {
+    tnsr::II<DataType, SpatialDim, Frame::Inertial> (*f)(
+        const tnsr::II<DataType, SpatialDim, Frame::Inertial>&,
+        const tnsr::I<DataType, SpatialDim, Frame::Inertial>&) =
+        &gr::transverse_projection_operator<SpatialDim, Frame::Inertial,
+                                            DataType>;
+    pypp::check_with_random_values<1>(f, "ProjectionOperators",
+                                      "transverse_projection_operator",
+                                      {{{-1., 1.}}}, used_for_size);
+  }
+
+  {
+    tnsr::ii<DataType, SpatialDim, Frame::Inertial> (*f)(
+        const tnsr::ii<DataType, SpatialDim, Frame::Inertial>&,
+        const tnsr::i<DataType, SpatialDim, Frame::Inertial>&) =
+        &gr::transverse_projection_operator<SpatialDim, Frame::Inertial,
+                                            DataType>;
+    pypp::check_with_random_values<1>(f, "ProjectionOperators",
+                                      "transverse_projection_operator",
+                                      {{{-1., 1.}}}, used_for_size);
+  }
+
+  {
+    tnsr::Ij<DataType, SpatialDim, Frame::Inertial> (*f)(
+        const tnsr::I<DataType, SpatialDim, Frame::Inertial>&,
+        const tnsr::i<DataType, SpatialDim, Frame::Inertial>&) =
+        &gr::transverse_projection_operator<SpatialDim, Frame::Inertial,
+                                            DataType>;
+    pypp::check_with_random_values<1>(
+        f, "ProjectionOperators",
+        "transverse_projection_operator_mixed_from_spatial_input",
+        {{{-1., 1.}}}, used_for_size);
+  }
+}
+}  // namespace
+
+namespace {
+using Affine = domain::CoordinateMaps::Affine;
+using Affine3D = domain::CoordinateMaps::ProductOf3Maps<Affine, Affine, Affine>;
+using frame = Frame::Inertial;
+constexpr size_t SpatialDim = 3;
+
+void test_spatial_projection_tensors_3D(
```

Comment:
It's not clear to me what this test function is adding over the python tests. Rather than comparing to a python implementation you seem to be comparing to values you worked out by hand or elsewhere? At least a short comment at the start of the function describing what the test is doing would be helpful

Reply:
This test is only comparing the output of these functions to values from spec. I've added comments to explain this.